services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -47,6 +48,7 @@
     private final Boolean consistentRead;
     private final Expression filterExpression;
     private final List<String> attributesToProject;
+    private final List<AttributeName> attributeNamesToProject;
```

Comment:
My opinion is that we should deprecate the List<String> and just have a List<AttributeName>. That will simplify our behavioral logic around this.